Mathlib/Topology/Algebra/LinearTopology.lean

```diff
@@ -154,6 +154,16 @@ theorem _root_.isLinearTopology_iff_hasBasis_open_submodule [ContinuousAdd M] :
       (fun N : Submodule R M ↦ IsOpen (N : Set M)) (fun N : Submodule R M ↦ (N : Set M)) :=
   ⟨fun _ ↦ hasBasis_open_submodule R, fun h ↦ .mk_of_hasBasis R h⟩
 
+theorem _root_.isLinearTopology_iff_exists_submodule_isOpen_subset [ContinuousAdd M] :
+    IsLinearTopology R M ↔
+      ∀ s : Set M, IsOpen s → 0 ∈ s → ∃ N : Submodule R M, IsOpen (X := M) N ∧ ↑N ⊆ s := by
+  rw [isLinearTopology_iff_hasBasis_open_submodule, hasBasis_iff]
+  refine ⟨fun H s hs h0s ↦ (H _).mp (hs.mem_nhds h0s), fun H t ↦ ?_⟩
+  refine ⟨fun ht ↦ ?_, fun ⟨N, hN, hNt⟩ ↦ mem_of_superset (hN.mem_nhds N.zero_mem) hNt⟩
+  obtain ⟨s, hst, hs, h0s⟩ := mem_nhds_iff.mp ht
+  obtain ⟨N, hN, hNs⟩ := H s hs h0s
+  exact ⟨N, hN, hNs.trans hst⟩
```

Comment:
I would like to avoid using the new lemma about open submodules.
```suggestion
  simp only [isLinearTopology_iff_hasBasis_submodule, hasBasis_iff]
  intro s
  refine ⟨fun h0s ↦ ?_, fun ⟨V, hV0, hVs⟩ ↦ mem_of_superset hV0 hVs⟩
  obtain ⟨V, hV⟩ := NonarchimedeanAddGroup.is_nonarchimedean s h0s
  exact ⟨V.1.toIntSubmodule, (isOpen_iff_mem_nhds.mp V.2) 0 V.zero_mem, hV⟩
```